src/static/riot/analytics/analytics.tag

```diff
@@ -0,0 +1,540 @@
+                    $(self.refs.date_selection_container).addClass('hidden')
+                }
+            })
+
+            /*---------------------------------------------------------------------
+             Calendar Setup
+            ---------------------------------------------------------------------*/
+
+            general_calendar_options = {
+                type: 'date',
+                // Sets the format of the placeholder date string to YYYY-MM-DD
+                formatter: {
+                   date: function (date, settings) {
+                       if (!date) return '';
+                       var day = date.getDate();
+                       var month = date.getMonth() + 1;
+                       var year = date.getFullYear();
+                       return year + '-' + month + '-' + day;
+                   }
+                },
+            }
+
+            start_specific_options = {
+                endCalendar: $(self.refs.end_calendar),
+                onChange: function(date, text) {
+
+                    let year = date.getFullYear()
+                    let month = date.getMonth()
+                    if(month === 0) {
+                        self.start_date = {
+                            year: year - 1,
+                            month: 12,
+                            day: 1,
+                        }
+                    } else {
+                        self.start_date = {
+                            year: year,
+                            month: month,
+                            day: 1,
+                        }
+                    }
+
+                    self.update_analytics(self.start_date, self.end_date, self.time_unit, false)
+                },
+            }
+
+            end_specific_options = {
+                startCalendar: $(self.refs.start_calendar),
+                onChange: function(date, text) {
+
+                    let year = date.getFullYear()
+                    let month = date.getMonth() + 1
+                    let day = date.getDate()
+                    self.end_date = {
+                        year: year,
+                        month: month,
+                        day: day,
+                    }
+
+                    self.update_analytics(self.start_date, self.end_date, self.time_unit, false)
+                },
+            }
+
+            start_calendar_options = _.assign({}, general_calendar_options, start_specific_options)
+            end_calendar_options = _.assign({}, general_calendar_options, end_specific_options)
+
+            $(self.refs.start_calendar).calendar(start_calendar_options);
+            $(self.refs.end_calendar).calendar(end_calendar_options);
+
+            /*---------------------------------------------------------------------
+             Chart Setup
+            ---------------------------------------------------------------------*/
+
+            var ctx_c = self.refs.competition_chart.getContext('2d');
```

Comment:
`var ctx_c = $(self.refs.competition_chart)` should be fine.

Reply:
(also, I believe `ctx_c` is from ChartsJS examples, but for us we should probably use `canvas` or `canvas_element` maybe, so it's clear at a glance in the future)

Reply:
Got rid of the variables. Uses jquery directly inside of the `new Chart()` calls.